vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -1,1017 +1,1023 @@
+        let firstPage = Math.floor(index / grid.pageSize);
+        let updatedPageCount = Math.ceil(length / grid.pageSize);
 
-      for (let i = 0; i < updatedPageCount; i++) {
-        let page = firstPage + i;
-        let items = cache[pkey][page];
-        grid.$connector.doDeselection(items.filter(item => selectedKeys[item.key]));
-        delete cache[pkey][page];
-        const updatedItems = updateGridCache(page, parentKey);
-        if (updatedItems) {
-          itemsUpdated(updatedItems);
+        for (let i = 0; i < updatedPageCount; i++) {
+          let page = firstPage + i;
+          let items = cache[pkey][page];
+          grid.$connector.doDeselection(items.filter(item => selectedKeys[item.key]));
+          delete cache[pkey][page];
+          const updatedItems = updateGridCache(page, parentKey);
+          if (updatedItems) {
+            itemsUpdated(updatedItems);
+          }
+          updateGridItemsInDomBasedOnCache(items);
         }
-        updateGridItemsInDomBasedOnCache(items);
-      }
-    };
+      });
 
-    const isSelectedOnGrid = function(item) {
-      const selectedItems = grid.selectedItems;
-      for(let i = 0; i < selectedItems; i++) {
-        let selectedItem = selectedItems[i];
-        if (selectedItem.key === item.key) {
-          return true;
-        }
-      }
-      return false;
-    }
-
-    grid.$connector.reset = function() {
-      grid.size = 0;
-      deleteObjectContents(cache);
-      deleteObjectContents(grid._cache.items);
-      deleteObjectContents(lastRequestedRanges);
-      if(ensureSubCacheDebouncer) {
-        ensureSubCacheDebouncer.cancel();
-      }
-      if(parentRequestDebouncer) {
-        parentRequestDebouncer.cancel();
-      }
-      ensureSubCacheDebouncer = undefined;
-      parentRequestDebouncer = undefined;
-      ensureSubCacheQueue = [];
-      parentRequestQueue = [];
-      updateAllGridRowsInDomBasedOnCache();
-    };
-
-    const deleteObjectContents = function(obj) {
-      let props = Object.keys(obj);
-      for (let i = 0; i < props.length; i++) {
-        delete obj[props[i]];
-      }
-    }
-
-    grid.$connector.updateSize = function(newSize) {
-      grid.size = newSize;
-    };
-
-    grid.$connector.updateUniqueItemIdPath = function(path) {
-      grid.itemIdPath = path;
-    }
-
-    grid.$connector.expandItems = function(items) {
-      let newExpandedItems = Array.from(grid.expandedItems);
-      items.filter(item => !grid._isExpanded(item))
-        .forEach(item =>
-          newExpandedItems.push(item));
-      grid.expandedItems = newExpandedItems;
-    }
-
-    grid.$connector.collapseItems = function(items) {
-      let newExpandedItems = Array.from(grid.expandedItems);
-      items.forEach(item => {
-        let index = grid._getItemIndexInArray(item, newExpandedItems);
-        if(index >= 0) {
-            newExpandedItems.splice(index, 1);
+      const isSelectedOnGrid = tryCatchWrapper(function(item) {
```

Comment:
This seems to only be called from within other methods that are already using `tryCatchWrapper` so it could be removed here.

Reply:
Updated.